runbotics/runbotics-desktop/src/action/zip/zip.test.ts

```diff
@@ -0,0 +1,282 @@
+                ZipActionHandler, 
+                ServerConfigService
+            ],
+        })
+            .overrideProvider(ServerConfigService)
+            .useValue({
+                tempFolderPath: TEMP_FOLDER_PATH,
+            })
+            .compile();
+        zipActionHandler = module.get(ZipActionHandler);
+        serverConfigService = module.get(ServerConfigService);
+    });
+
+    beforeEach(async () => {
+        if (!fs.existsSync(TEMP_FOLDER_PATH)) {
+            fs.mkdirSync(TEMP_FOLDER_PATH, { recursive: true });
+        }
+
+        if (fs.existsSync(`${ZIP_TESTING_FOLDER}.zip`)) {
+            fs.rmSync(`${ZIP_TESTING_FOLDER}.zip`);
+        }
+    });
+
+    afterEach(async () => {
+        if (fs.existsSync(ZIP_TESTING_FOLDER)) {
+            fs.rmSync(ZIP_TESTING_FOLDER, { recursive: true });
+        }
+
+        if (fs.existsSync(`${TEMP_FOLDER_PATH}.zip`)) {
+            fs.rmSync(`${TEMP_FOLDER_PATH}.zip`, { recursive: true });
+        }
+
+    });
+
+    afterAll(async () => {
+        if (fs.existsSync(serverConfigService.tempFolderPath)) {
+            fs.rmdirSync(serverConfigService.tempFolderPath, { recursive: true });
+        }
+
+        if (fs.existsSync(ZIP_TESTING_FOLDER)) {
+            fs.rmdirSync(ZIP_TESTING_FOLDER);
+        }
+
+        if (fs.existsSync(TEMP_FOLDER_PATH)) {
+            fs.rmdirSync(TEMP_FOLDER_PATH, { recursive: true });
+        }
+    });
+
+    const createTestFolder = (folderPath: string) => {
+        if (!fs.existsSync(folderPath)) {
+            fs.mkdirSync(folderPath, { recursive: true });
+        }
+    };
+
+    const createTestZip = async (fileName: string, filePath?: string, zipName?: string) => {
+        const toZipPath = filePath ? `${filePath}${path.sep}${fileName}` : `${TEMP_FOLDER_PATH}${path.sep}${fileName}`;
+        createTestFolder(toZipPath);
+        zipActionHandler.zipFile({ fileName, path: filePath, zipName});
+    };
+
+    const removeTestFolder = (folderPath: string) => {
+        if (fs.existsSync(folderPath)) {
+            fs.rmSync(folderPath, { recursive: true });
+        }
+    };
+
+    it('should be defined', () => {
+        expect(zipActionHandler).toBeDefined();
+        expect(serverConfigService).toBeDefined();
+    });
+
+    describe('Create archive', async() => {
+        it('Should create zip named with fileName when optional inputs (path and zipName) are not provided', async() => {
+            const fileName = 'zipArchiveName';
+            const params: ZipFileActionInput = {
+                fileName,
+                path: undefined,
+                zipName: undefined
+            };
+
+            createTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${fileName}`);
+            await zipActionHandler.zipFile(params);
+            expect(
+                fs.existsSync(`${TEMP_FOLDER_PATH}${path.sep}${fileName}.zip`)
+            ).toBeTruthy();
+            removeTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${fileName}`);
```

Comment:
if the new tests structure will be implemented (/temp/zip-tests/ instead of just /temp) it's good to think ahead - after the changing structure we don't want to refactor the whole file - maybe a variable will be enough, maybe a function similar to resolvePath will be needed